src/coreclr/src/debug/daccess/dacfn.cpp

```diff
@@ -277,9 +277,12 @@ DacVirtualUnwind(ULONG32 threadId, PT_CONTEXT context, PT_KNONVOLATILE_CONTEXT_P
     {
         hr = dt->VirtualUnwind(threadId, sizeof(CONTEXT), (BYTE*)context);
     }
-    else
 #endif
+
+    if (hr == E_NOINTERFACE || hr == E_NOTIMPL)
```

Comment:
This looks fishy.  What if `!FEATURE_DATATARGET4` and `SUCCEEDED(hr)`

Reply:
Then hr is E_NOINTERFACE and PAL_VirtualUnwindOutOfProc is called.

What SUCCEEDED(hr) are you looking at? 

```

    HRESULT hr = E_NOINTERFACE;           

#ifdef FEATURE_DATATARGET4
    ReleaseHolder<ICorDebugDataTarget4> dt;
    hr = g_dacImpl->m_pTarget->QueryInterface(IID_ICorDebugDataTarget4, (void **)&dt);
    if (SUCCEEDED(hr))
    {
        hr = dt->VirtualUnwind(threadId, sizeof(CONTEXT), (BYTE*)context);
    }
#endif

    if (hr == E_NOINTERFACE || hr == E_NOTIMPL)
    {
        hr = S_OK;

        SIZE_T baseAddress = DacGlobalBase();
        if (baseAddress == 0 || !PAL_VirtualUnwindOutOfProc(context, contextPointers, baseAddress, DacReadAllAdapter))
        {
            hr = E_FAIL;
        }
    }
```